internal/domain/utils.go

```diff
@@ -0,0 +1,50 @@
+package domain
+
+import (
+	"math/rand"
+	"unicode"
+)
+
+type SignCount struct {
+	upperCaseCount int
+	lowerCaseCount int
+	digitCount     int
+}
+
+func charTypeCounts(password string) SignCount {
+	var upperCaseCount, lowerCaseCount, digitCount int
+
+	for _, char := range password {
+		switch {
+		case unicode.IsUpper(char):
+			upperCaseCount++
+		case unicode.IsLower(char):
+			lowerCaseCount++
+		case unicode.IsDigit(char):
+			digitCount++
+		}
+	}
+
+	return SignCount{
+		upperCaseCount: upperCaseCount,
+		lowerCaseCount: lowerCaseCount,
+		digitCount:     digitCount,
+	}
+}
```

Comment:
Suggestion to make it a bit more clean.
```go
func charTypeCounts(password string) SignCount {
	counts := SignCount{}

	for _, char := range password {
		switch {
		case unicode.IsUpper(char):
			counts.upperCaseCount++
		case unicode.IsLower(char):
			counts.lowerCaseCount++
		case unicode.IsDigit(char):
			counts.digitCount++
		}
	}

	return counts
}
```

Reply:
You're right, it's much better, I didn't know it was possible to create an object like that in GO without the required parameters :D. Fixed :)

Reply:
@Vaserty FYI all structs can be created like that, no matter what. Go operates with zero-values by default. So you can instantiate any struct and if you don't pass the value for some fields, they will be filled with zero-value for a given type; so for `int` it will be 0, for `bool` it will be false, for `string` it will be "", etc.

Reply:
thanks ! :)